src/entrypoint.sh

```diff
@@ -155,20 +169,40 @@ function git_init() {
     mkdir -p "${HOME}"/.ssh
     ssh-keyscan -t rsa "${source_repo_hostname}" >> "${HOME}"/.ssh/known_hosts
   else
-    info "the source repository is located within GitHub."               
-    if [[ -n "${SOURCE_GH_TOKEN}" ]]; then      
-      unset GITHUB_TOKEN      
-      gh auth login --git-protocol "https" --hostname "${SOURCE_REPO_HOSTNAME}" --with-token <<< "${SOURCE_GH_TOKEN}"        
+    info "the source repository is located within GitHub."
+    # GITHUB_TOKEN is deprecated and can be removed in the future
+    if [[ -n "${SOURCE_GH_TOKEN}" ]] || [[ -n "${GITHUB_TOKEN}" ]]; then
+      ################################
+      if [[ -n "${GITHUB_TOKEN}" ]]; then
```

Comment:
the idea is to have a migration path and to not break current implementations.
If the env vatiable is set, this should not break the action